impeller/renderer/backend/vulkan/render_pass_vk.cc

```diff
@@ -368,70 +370,41 @@ void RenderPassVK::SetInstanceCount(size_t count) {
   instance_count_ = count;
 }
 
-static bool AppendVertexBuffer(
-    CommandBufferVK& command_buffer,
-    std::vector<vk::Buffer>& vertex_buffers,
-    std::vector<vk::DeviceSize>& vertex_buffer_offsets,
-    const BufferView& vertex_buffer_view) {
-  if (!vertex_buffer_view) {
+// |RenderPass|
+bool RenderPassVK::SetVertexBuffer(BufferView vertex_buffers[],
+                                   size_t vertex_buffer_count,
+                                   size_t vertex_count) {
+  if (!ValidateVertexBuffers(vertex_buffers, vertex_buffer_count)) {
     return false;
   }
-  auto vertex_buffer = vertex_buffer_view.buffer;
 
-  if (!vertex_buffer) {
-    VALIDATION_LOG << "Failed to acquire device buffer"
-                   << " for vertex buffer view";
-    return false;
+  vk::Buffer buffers[kMaxVertexBuffers];
+  vk::DeviceSize vertex_buffer_offsets[kMaxVertexBuffers];
+  for (size_t i = 0; i < vertex_buffer_count; i++) {
+    buffers[i] = DeviceBufferVK::Cast(*vertex_buffers[i].buffer).GetBuffer();
```

Comment:
We also need to track each buffer (at least until I refactor us to not need the auto tracking)

Reply:
Whoops, nice catch. Done.